### PR TITLE
Fixed #20597 -- Replaced admin GIF/PNG icons by SVG

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/20597

todo:
- [x] update tests for new icon names
- [x] update the "no" icon as done in https://github.com/elky/django-flat-theme/pull/14
- [x] Add a mention in the 1.9 release notes

### CODE REVIEW

**Comment by timgraham** [comment]
@elky, I created this pull request and added a few todos in the description based on things I could think of. Any other blockers you can think of? p.s. how are you authoring/editing the icons?

**Comment by elky** [comment]
@timgraham thanks for that. One only blocker is that need to refactor couple of things there -- so don't merge it yet.

Regarding authoring/editing. All these icons are from Font Awesome pack. I just found a [repo](https://github.com/encharm/Font-Awesome-SVG-PNG/tree/master/black/svg) with SVG originals and used icons I needed. Font Awesome fonts are under OFL licence -- so I think it's OK that we can use some of icons from the pack. Correct me if I'm wrong.

**Comment by timgraham** [comment]
"Font-Awesome-SVG-PNG is licensed under the MIT license" so I think we just need to include the MIT License in the svg folder? Maybe the OFL license too?

**Comment by elky** [comment]
Yeah, I'll put both.

**Comment by elky** [comment]
@timgraham I updated tests with new icons, but something wrong there. Can you please check it?

**Comment by elky** [comment]
Here are few screenshots from IE8:

django-1-ie8
django-2-ie8
django-3-ie8

As you can see there are few places where accessibility is broken -- alt text works for `img` tag only, so some icons (calendar, clock, selector arrows) are just empty which looks weird. 

@timgraham are you sure we drop IE8 hard?

**Comment by timgraham** [comment]
The question of IE8 support has been submitted to the technical board for a vote, we should hear back shortly.

The "checkout" errors on Jenkins are nothing to worry about. It happens on occasion.

**Comment by elky** [comment]
OK, I mentioned SVG icons in release notes, waiting for IE8 decision.

File: docs/releases/1.9.txt [hunk 1]
```diff
@@ -118,9 +118,10 @@ though:
 New styling for ``contrib.admin``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The admin sports a modern, flat design. It still provides a fully-functional
-experience to `YUI's A-grade`_ browsers. Older browser may experience varying
-levels of graceful degradation.
+The admin sports a modern, flat design with new SVG icons which look perfect
+on hi-dpi screens. It still provides a fully-functional experience to
```
Comment (timgraham) [inline docs/releases/1.9.txt]: HiDPI?

Comment (elky) [inline docs/releases/1.9.txt]: Oops, fixed.

**Comment by timgraham** [comment]
Looks good to me, and I just posted to django-developers that the technical board approved dropping IE8 support. Anything else needed from your perspective?

**Comment by elky** [comment]
No, all done. Thanks @timgraham

**Comment by timgraham** [comment]
Oh, how about removing the old, unused icons? Can we do anything about move_vertex_on.png?

**Comment by elky** [comment]
I think we should keep old icons because some apps use them. Regarding some really old things - I'm going to make another PR with refactoring, just found lots of code in CSS that not used anywhere. So I'll clean it up with unused old icons as well. Is it fine?

**Comment by timgraham** [comment]
I think if other apps use the icons, they can create an app with them without much difficulty I don't see a need to retain them in Django indefinitely, but sure, we can remove them in a separate PR since we already still have the ones dropped in the redesign.

Would still be nice to replace `move_vertex_on.png`/`move_vertex_off.png` with SVG if possible in this commit I think (the icon seems a bit specialized though, not sure we can find an equivalent in font awesome).

**Comment by elky** [comment]
> they can create an app with them

If is normal practice? I mean usually Django warns developers about soon deprecations. I saw couple of CMS apps and file managers that use default icons - so they will have to update their code in rush. 

> move_vertex_on.png/move_vertex_off.png

Could you please say where can I see these icons in action?

**Comment by timgraham** [comment]
I don't consider the icons a "stable API" such that we need to maintain backwards compatibility. I suspect other projects will want to update their apps to look nice with the new admin theme anyway, no?

You can see the vertext png if you work through the [GeoDjango tutorial](https://docs.djangoproject.com/en/dev/ref/contrib/gis/tutorial/), but actually it looks like we can just leave them as-is as there seems to be other icons fetched from openlayers.org which use similar theming.

**Comment by timgraham** [comment]
merged in c32b61c6fd6ddf39fd49585954d422ef8d7304f6, thanks!

**Comment by elky** [comment]
@timgraham thanks for clarification regarding geo -- I quickly draw svg icons for it:
- https://github.com/elky/django-flat-theme/blob/master/flat/static/admin/img/gis/move_vertex_on.svg
- https://github.com/elky/django-flat-theme/blob/master/flat/static/admin/img/gis/move_vertex_off.svg

I'll put them in my next PR.

**Pull request closed without merging**